packages/core/src/common/utils/safeInvokeMember.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the terms of the LICENSE file distributed with this project.
+ */
+
+import { isFunction } from "../utils";
+
+/**
+ * Safely invoke the member function with no arguments, if the object
+ * exists and the given key is indeed a function, and return its value.
+ * Otherwise, return undefined.
+ */
+export function safeInvokeMember<T extends { [P in K]?: () => R }, K extends keyof T, R = void>(
```

Comment:
Where is this `P` used? Guess I don't quite follow the typing foo here.

Reply:
added big `/* comment */` explaining types. hope that helps!

Reply:
Thanks!